fili-core/src/main/java/com/yahoo/bard/webservice/application/AbstractBinderFactory.java

```diff
@@ -635,7 +636,7 @@ protected DruidFilterBuilder buildDruidFilterBuilder() {
      * @return An object to generate having maps from having string.
      */
     protected HavingGenerator buildHavingGenerator(ConfigurationLoader loader) {
-        return new DefaultHavingApiGenerator(loader);
+        return new PerRequestDictionaryHavingGenerator(new DefaultHavingApiGenerator(loader.getMetricDictionary()));
```

Comment:
Looks like we don't need to pass `loader` anymore, but only `MetricDictionary` instead

Reply:
Two reasons not to:

1. Just because *this* `HavingGenerator` doesn't need all the configuration, doesn't mean other people's custom `HavingGenerator won't.

2. Changing `buildHavingGenerator` would break any customers that override it, and we don't want to make backwards incompatible changes if we don't have to.

Reply:
make sense